utils/python/tests/scripts_regression_tests

```diff
@@ -134,30 +134,50 @@ class B_TestCreateNewcase(unittest.TestCase):
 ###############################################################################
     def setUp(self):
         self._testroot = MACHINE.get_value("CESMSCRATCHROOT")
-        self._testdir = os.path.join(self._testroot, 'scripts_regression_tests.testcreatenewcase')
-        if os.path.exists(self._testdir):
-            shutil.rmtree(self._testdir)
-        self._do_teardown = False
+        self._testdirs = []
+        self._do_teardown = []
 
     def test_createnewcase(self):
-        cmd = "%s/create_newcase --case %s --compset X --res f19_g16" % (SCRIPT_DIR, self._testdir)
+        testdir = os.path.join(self._testroot, 'scripts_regression_tests.testcreatenewcase')
+        if os.path.exists(testdir):
+            shutil.rmtree(testdir)
+        self._testdirs.append(testdir)
+        cmd = "%s/create_newcase --case %s --compset X --res f19_g16" % (SCRIPT_DIR, testdir)
         stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=SCRIPT_DIR)
         self.assertEqual(stat, 0, msg="COMMAND '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, output, errput))
         cmd = "./case.setup "
-        stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=self._testdir)
-        self.assertEqual(stat, 0, msg="COMMAND '%s' from case directory '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, self._testdir, output, errput))
+        stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=testdir)
+        self.assertEqual(stat, 0, msg="COMMAND '%s' from case directory '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, testdir, output, errput))
         cmd = "./case.build"
-        stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=self._testdir)
-        self.assertEqual(stat, 0, msg="COMMAND '%s' from case directory '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, self._testdir,output, errput))
-        self._do_teardown = True
+        stat, output, errput = run_cmd(cmd, ok_to_fail=True, from_dir=testdir)
+        self.assertEqual(stat, 0, msg="COMMAND '%s' from case directory '%s' SHOULD HAVE WORKED\noutput:\n%s\n\nerrput:\n%s" % (cmd, testdir,output, errput))
+        self._do_teardown.append(testdir)
+
+    def test_user_mods(self):
```

Comment:
Add a test for user_mods - this test could be improved it doesn't yet test the shell_commands capability.